Marlin/src/module/planner.cpp

```diff
@@ -853,8 +859,8 @@ void Planner::calculate_trapezoid_for_block(block_t * const block, const_float_t
   #endif
 
   // Store new block parameters
-  block->accelerate_until = accelerate_steps;
-  block->decelerate_after = block->step_event_count - decelerate_steps;
+  block->accelerate_before = accelerate_steps;
+  block->decelerate_start = block->step_event_count - decelerate_steps;
```

Comment:
To reiterate the rationale for renaming these, even before this PR they needed a good stern renaming. The number of `accelerate_steps` is one more than the index at which acceleration ends (i.e., the index at which cruising starts). And `block->step_event_count - decelerate_steps` gives us the step index at which deceleration starts.

Reply:
I don't quite understand what you mean so instead I'll just bring up the fencepost error. Where if you have a 5 segment fence you need 6 fence posts. Say you carry over the same thinking in steps and acceleration steps in between (because they're changes in speed) and again the same thinking for deceleration, then again realizing that a segment can be full acceleration, full deceleration, accel + decel or accel + cruise + decel. That's a lot of cases to keep thinking about and perfectly match the indices (fenceposts) to the acceleration/deceleration in between. Then add the conversion from float into distinct int steps.

It's easier to switch the mental model to steps representing the segments centered on them. Then 5 fence segments 5 fence posts (centered).

Reply:
I really do not yet see a valid reason to rename the vars here, they are still ambiguous before and after the renaming. I think it just adds more confusion to the former state. From my perception they are  both wrong and right at the same time based on how you wish to interpret the index. If we wish to clear the meaning or "intent" up then we need to clearly doc the index method and the what comparison point of the count to the index outcome should be.

Reply:
If we wish to rename for clarity then possibly we can keep a context of the var use. 
e.g.
accelerate_stop_point
decelerate_start_point

The index count position is observed and we would to adjust the comparison logic to meet the correct PIT for the actual outcomes. We would have to comment what the comparison logic intent is at every point. 

Another context view e.g.:
accelerate_count_end
decelerate_count_begin

Reply:
There are some number of steps (`S`) in a given linear move. Some number of those steps are acceleration (`A`), cruising (`C`) and deceleration (`D`), so we store the indices in the linear move that correspond to a zero-based step index (`I`) where cruising starts and where deceleration starts.

The fencepost analogy only seems to make sense if the starting-point and the ending-point are included in the count. But they're not. In the linear move code, the first step is indexed as 0, but it does not represent fencepost 0. It represents the step that starts at fencepost 0 and ends at fencepost 1. So a complete linear move starts with the  fencepost 0->1 step and the last step is fencepost `S-1`->`S`, at which time the index is `S-1`, and only after which the index is `S`. When the step counter `I` is equal to `S` it has previously processed all steps.

The names suggested by @descipher miss that the unit is "steps" and not "points" and we're not so much "counting" as we are "tracking." These variables are zero-based step indexes, being compared to a zero-based step counter. Variables with "count" tend to count things where we care about "how many," while variables with "point" tend to be something like an `xyz_pos_t`.

If there is some confusion that "step" in these contexts represents a "step index" then we may add "`index`" to the name, but it should still refer to the unit of a "step."